### PR TITLE
[fga] don't await spicedb calls when disabled

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Don't await spicedb checks and migrate user unless checks are enabled.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 34f9094</samp>

This pull request improves the performance and reliability of the new authorization system based on SpiceDB. It adds feature flag checks to avoid unnecessary data migrations and SpiceDB calls, and handles errors gracefully in the `RelationshipUpdater` and `Authorizer` classes. It also refactors the `checkPermission` method of the `SpiceDBAuthorizer` class to simplify the logic.

</details>

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-spicedb-async</li>
	<li><b>🔗 URL</b> - <a href="https://se-spicedb-async.preview.gitpod-dev.com/workspaces" target="_blank">se-spicedb-async.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - se-spicedb-async-gha.17673</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-se-spicedb-async%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/authorization/relationship-updater.ts [hunk 1]
```diff
@@ -39,6 +39,24 @@ export class RelationshipUpdater {
      * @returns
      */
     public async migrate(user: User): Promise<User> {
+        const isChecksEnabled = await isFgaChecksEnabled(user.id);
+        const migrated = this.internalMigrate(user);
+        // if checks are not enabled, we don't want to wait for the migration to finish
+        if (!isChecksEnabled) {
```
Comment (geropl) [inline components/server/src/authorization/relationship-updater.ts:45]: :heavy_check_mark:

File: components/server/src/authorization/relationship-updater.ts [hunk 2]
```diff
@@ -39,6 +39,24 @@ export class RelationshipUpdater {
...
+        } catch (error) {
+            log.error({ userId: user.id }, "Error while migrating user", error);
+            throw error;
+        }
+    }
+
+    private async internalMigrate(user: User): Promise<User> {
         const isEnabled = await isFgaWritesEnabled(user.id);
         if (!isEnabled) {
             if (user.additionalData?.fgaRelationshipsVersion !== undefined) {
```
Comment (geropl) [inline components/server/src/authorization/relationship-updater.ts:62]: nit: would still love a explicit "Figration failed" catch block in internalMigrate.

Comment (svenefftinge) [reply components/server/src/authorization/relationship-updater.ts:62]: I added it to the migrate method

**Comment by geropl** [comment]
Code LGTM :heavy_check_mark: 

The test failure is due to the new "write flag" handling.

Waiting for the preview env to test.

File: components/server/src/authorization/relationship-updater.spec.db.ts [hunk 3]
```diff
@@ -79,6 +79,9 @@ describe("RelationshipUpdater", async () => {
         });
 
         user = await migrator.migrate(user);
+
+        // the update is not sync so we sleep a sec
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (geropl) [inline components/server/src/authorization/relationship-updater.spec.db.ts:84]: Uh, hopefully this is not too flaky :crossed_fingers:

**Pull request merged**